### DISCUSSION TITLE
Rate limiting for Storage and Image Transformations API calls

### DISCUSSION THREAD

**Discussion by arbyruns** [supabase/supabase#29108] [Feature Requests] [answered]
# Bug report

- [X] I confirm this is a bug with Supabase, not with my own application. 
- [X] I confirm I have searched the [Docs](https://docs.supabase.com), GitHub [Discussions](https://github.com/supabase/supabase/discussions), and [Discord](https://discord.supabase.com). 

## Describe the bug

Currently, there is no ability to disable storage image transformations and if a user is able to determine where images resides can make repeated image transformation calls. An unknown actor can then make repeated calls churning through usage.

## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

1. Have images reside in storage and send a transofrmation API call.
2. An example of the URL being sent to my project https://<id>.supabase.co/storage/v1/render/image/public/<location>/<image>.jpg?width=1200&height=1200&resize=contain&quality=80

## Expected behavior

Ability to disable this feature or block those API calls.

## Additional context

Support has informed me that there is no way to disable this feature.

**Comment by encima** [comment, marked as answer]
Hey there,
Right now, the configuration of those endpoints are not exposed but we can take this as a feature request! I will convert this to a Discussion and move it to feature requests!

When self-hosting, you can modify the docker compose config to include the `rate-limiting` plugin for the `kong` container.

**Comment by imagitama** [comment]
In my application I was using the image transformation endpoint but because I was overusing it, I switched to optimisation on upload.

It says I am still using the transformation endpoints. It would be nice to be able to disable this completely.
